rp2040-hal/src/spi.rs

```diff
@@ -263,22 +285,26 @@ impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
     }
 
     /// Initialize the SPI in master mode
-    pub fn init<F: Into<HertzU32>, B: Into<HertzU32>, M: Into<Mode>>(
+    pub fn init<F: Into<HertzU32>, B: Into<HertzU32>>(
```

Comment:
Why removing the generic over Mode instead of updating it?
```suggestion
    pub fn init<F: Into<HertzU32>, B: Into<HertzU32, M: Into<FrameFormat>>>(
```